### ISSUE TITLE
only allow user-agents via regex?

### ISSUE THREAD

**Issue by bogorad** [agence-gaya/traefik-plugin-blockuseragent#5]
I have a need to only allow one specific crawler, idenfitied by its user-agent, specified in via a regex. Can this be implemented?

**Comment by jliebert** [comment]
It's already, you can just add one regex if you want.

**Comment by bogorad** [comment]
I don't understand - in the readme it says that the regex defines the user-agent that it to be **blocked**. I need the opposite function, block all user-agents **except** the one I specify via regex.

**Comment by jliebert** [comment]
Do a negation in your regex: ^(?!bar$).*$

**Comment by bogorad** [comment]
I'm probably being slow, sorry about that in advance, but while this works as expected:

```
  middlewares:
    mw_block_user_agnet:
      plugin:
        blockUserAgent:
          regex:
            - "^(aaa).*$"
```

this gives 404 (!) regardless of the contents of user-agent string.

```
  middlewares:
    mw_block_user_agnet:
      plugin:
        blockUserAgent:
          regex:
            - "^(?!aaa).*$"
```

Any ideas?

**Comment by jliebert** [comment]
I will do test on my side.
My apologies, revers regex is not realy a solution if you need long list of allowed user agent. 
I have reopen your case and will studies on it.

**Comment by bogorad** [comment]
I don't need a long list, maybe 60 chars max.

**Comment by jliebert** [comment]
@bogorad, the go regex compiler don't support Perl syntax `(?!`.

I propose to add a new list regexAllow, if a user-agent match the list it is allowed, this look like this:

```
middlewares:
    allow-only-agent-aaa:
       plugin:
         blockUserAgent:
           regexAllow:
              - '^aaa$'
           # deny all other
           regex:
              - '.*'
```

**Comment by jliebert** [comment]
I have published release v0.1.8, it should do what you need.

**Comment by bogorad** [comment]
Works beautifully, thank you so much!
